fetch.bs

```diff
@@ -2204,6 +2204,39 @@ unset or <a for=request>keepalive</a> is false, <a lt=terminated for=fetch>termi
 <a for="fetch group">fetch record</a>'s <a for="fetch record">fetch</a>.
 
 
+<h3 id=resolving-domains>Resolving domains</h3>
+
+<p>To <dfn>resolve a domain</dfn>, given a <a for=/>domain</a> <var>domain</var> and a
+<a for=/>network partition key</a> <var>key</var>, perform an <a>implementation-defined</a>
+operation to turn <var>domain</var> into a <a for=/>set</a> of one or more
+<a for=/>IP addresses</a>. If this operation succeeds, return the <a for=/>set</a> of
+<a for=/>IP addresses</a>. If it fails, return failure. The results of this operation may be cached.
+If they are cached, <var>key</var> must be used as part of the cache key.
```

Comment:
I'm not sure this line makes sense, or at least, not normative sense.

The issue here is that unless you're writing your own recursive resolver implementation, the resolvers you communicate with will ignore `key` and cache. E.g. if you use an OS API and the OS API caches, it's not going to consider `key`. And even if you write your own recursive resolver, you can't be sure that the name servers themselves won't cache based on `(host, recordtype)` tuples, ignoring `key` (which is never on the wire)

While this is definitely more of @MattMenke2 's realm for wording, I wanted to flag this, because it seems like this might involve some form of note or form of [RFC 6919 language](https://datatracker.ietf.org/doc/html/rfc6919#section-1).

Reply:
I noticed this too, but wasn't sure it was worth commenting on, given the style here tends to be not to explain things.  Anyhow, I'm not sure if it's reasonable to note that upstream DNS servers may still cache across keys, leave as-is, or remove the section.

Reply:
That does seem good to point out. And it seems that even for local lookup this can at best be a SHOULD for now.

Reply:
I think this is good now.

Reply:
@MattMenke2 FWIW, I'd be happy to explain more or accept patches that explain more of it, but as I understand it from @sleevi that would be quite a bit of effort and it's not clear it'll help solve a problem. But we do need some basics written down for the rest of the things to make sense and fit together. In general though I'd say that the style is the opposite of what you suggest, in that we do try to cover all the details.

Reply:
The style tends to be purely descriptive of what browsers do, with no additional detail (e.g., nowhere is it mentioned why network partition keys exist, or what the "essence" of a MIME type is intended for, merely how its calculated and used).

Reply:
Ah, I see what you mean. That's not really intentional, more of a side effect of prioritizing the normative description and having limited time. 😟